plugins/studymaster-web/src/main/java/de/dfki/vsm/xtension/studymasterWeb/WebStudyMasterExecutor.java

```diff
@@ -0,0 +1,306 @@
+    private final ArrayList<WsConnectContext> mWebsockets = new ArrayList<>();
+    /** The Javalin HTTP server. */
+    private Javalin mHttpServer;
+
+    private String mRequestResultVar ;
+
+    private String mSceneflowGoVar ;
+
+    /** This is the name of a Boolean global project variable that will be bet to _true_ when at least one web app is connected.*/
+    public String mGUIConnectedVar = "";
+
+    /** This is a "cache" of the last request sent to all connected GUIs.
+     * The string contains the message already formatted to the communication protocol.
+     * It will be used to send again the request to GUIs connecting after the REQUEST action was issued.
+     * If null, no pending request is active.
+     */
+    private String mLastRequestMessage = null;
+
+    /**
+     * Default constructor, pass values to superclass.
+     *
+     * @param config  Plugin configuration, as it can be set via the extensions settings, or manually via xml.
+     * @param project The running project the plugin is applied to.
+     */
+    public WebStudyMasterExecutor(PluginConfig config, RunTimeProject project) {
+        super(config, project);
+    }
+
+
+    @Override
+    public synchronized String marker(long id) {
+        return "$(" + id + ")";
+    }
+
+
+    @Override
+    public void launch() {
+        mLogger.message("Loading StudyMaster message sender and receiver ...");
+
+        //
+        // Retrieve property values
+        mGUIConnectedVar = mConfig.getProperty(sGUI_CONNECTED_VAR, sGUI_CONNECTED_VAR_DEFAULT);
+        final int http_port = Integer.parseInt(Objects.requireNonNull(mConfig.getProperty(sJAVALIN_PORT_VAR, sJAVALIN_PORT_VAR_DEFAULT)));
+        mRequestResultVar = mConfig.getProperty(sREQUEST_RESULT_VAR, sREQUEST_RESULT_VAR_DEFAULT) ;
+        mSceneflowGoVar = mConfig.getProperty(sGO_VAR, sGO_VAR_DEFAULT);
+
+        // Start the HTTP server
+        mHttpServer = Javalin.create(config -> {
+            config.addStaticFiles("/react-studymaster/build");
+            config.enforceSsl = true;
+        }).start(http_port);
+
+        // Set callbacks to manage WebSocket events
+        mHttpServer.ws("/ws", ws -> {
+            ws.onConnect(this::addWs);
+            ws.onMessage(ctx -> this.handleGUIMessage(ctx.message()));
+            ws.onClose(this::removeWs);
+            ws.onError(ctx -> mLogger.failure("WebSocket Error: " + ctx.error()));
+        });
+    }
+
+
+    @Override
+    public void unload() {
+        for (WsConnectContext ws: mWebsockets) {
+            if (ws.session.isOpen()) {
+                ws.session.close();
+            }
+        }
+        mWebsockets.clear();
+        mHttpServer.stop();
+    }
+
+
+    /** Invoked when a new websocket connection is opened (new web app is loaded).*/
+    private synchronized void addWs(WsConnectContext ws) {
+        mLogger.message("New WebSocket connection.");
```

Comment:
Maybe change the log level to debug, this could become noisy

Reply:
There is not `debug()` method for the logger. Did you men `warning()` ?